File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/interactive/skill/SkillActionDescription.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait SkillActionDescription extends TypeDeserializer {
  val typeId = 102

  protected trait Type extends BaseType with Serializable {
    val typeId = SkillActionDescription.typeId

    val skillId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(skillId)
    }

  }

  protected def create(skillId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val skillId = buf.readShort()

    create(skillId)
  }

}

object SkillActionDescription extends SkillActionDescription {
  case class TypeImpl(skillId: Short) extends Type
  override def create(skillId: Short) = TypeImpl(skillId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(skillId: Short) = create(skillId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/SkillActionDescription.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class SkillActionDescription extends NetworkType {
    public static final short TYPE_ID = 102;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short skillId;
    
    public SkillActionDescription() { }
    
    public SkillActionDescription(short skillId) {
        this.skillId = skillId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(skillId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        skillId = buf.readShort();
        if (skillId < 0)
            throw new RuntimeException("Forbidden value on skillId = " + skillId + ", it doesn't respect the following condition : skillId < 0");
    }
    
}
